### PR TITLE
Wildsearch

### PR DESCRIPTION
**Pull request by andrewfraser73**
Ability to search change requests using wildcards either * or % work
The following work
- TESTASS3
- TESTASS*
- *3
- *LEWIS*

Would have liked a LINQ like that worked with EF but Regex not supported and hence clunky if to cope with these conditions and can't cope with something like PCAS */L1* but that's an edge case

also added a filter with ability to show recent live changes but will need Jacki to see if these values are useful or not

### CODE REVIEW

File: src/Facade/Services/ChangeRequestFacadeService.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
     using System.Collections.Generic;
     using System.Linq;
     using System.Linq.Expressions;
+    using System.Text.RegularExpressions;
```
Comment (richardiphillips) [inline src/Facade/Services/ChangeRequestFacadeService.cs:7]: not being used if you've abandoned regex?

File: src/Facade/Services/ChangeRequestFacadeService.cs [hunk 2]
```diff
@@ -195,6 +196,59 @@ public IResult<IEnumerable<ChangeRequestResource>> GetChangeRequestsRelevantToBo
...
 
+        public IResult<IEnumerable<ChangeRequestResource>> SearchChangeRequests(
+            string searchTerm,
+            bool? outstanding, 
+            int? lastMonths,
+            IEnumerable<string> privileges = null)
+        {
+            var changeRequests = new List<ChangeRequest>();
+
+            var fromDate = (lastMonths == null)
```
Comment (richardiphillips) [inline src/Facade/Services/ChangeRequestFacadeService.cs:207]: could move into relevant part of IF statement. Also the hard coded date will date quickly. probably better to be relative to today

File: src/Facade/Services/ChangeRequestFacadeService.cs [hunk 3]
```diff
@@ -195,6 +196,59 @@ public IResult<IEnumerable<ChangeRequestResource>> GetChangeRequestsRelevantToBo
...
+                               ? new DateTime(2020, 1, 1)
+                               : DateTime.Now.AddMonths(-1 * (int)lastMonths);
+            var inclLive = ((outstanding == false) ? "LIVE" : "JUSTOUTSTANDING");
```
Comment (richardiphillips) [inline src/Facade/Services/ChangeRequestFacadeService.cs:210]: feels like too many brackets

File: src/Facade/Services/ChangeRequestFacadeService.cs [hunk 4]
```diff
@@ -195,6 +196,59 @@ public IResult<IEnumerable<ChangeRequestResource>> GetChangeRequestsRelevantToBo
...
+
+            var newPartNumber = searchTerm.Trim().ToUpper().Replace('%', '*');
+            var partSearch = newPartNumber.Split('*');
+
+            // this big if is just because Linq/EF/Oracle doesn't do a LIKE
+            // supports IC*, *3, *LEWIS*, PCAS*L1R1 but not multiple * e.g. PCAS */L1*
+            if (string.IsNullOrEmpty(newPartNumber))
+            {
+                var a = this.repository.FilterBy(
```
Comment (richardiphillips) [inline src/Facade/Services/ChangeRequestFacadeService.cs:219]: is this just testing code that can be removed?

**Review by richardiphillips — APPROVED** [review]

File: src/Facade/Services/ChangeRequestFacadeService.cs [hunk 5]
```diff
@@ -253,5 +307,7 @@ protected override void DeleteOrObsoleteResource(ChangeRequest entity, IEnumerab
         {
             throw new NotImplementedException();
         }
+
+
```
Comment (lewisrenfrew) [inline src/Facade/Services/ChangeRequestFacadeService.cs:311]: blankity blank

**Pull request merged**